lib/compat/wordpress-6.2/class-gutenberg-rest-pattern-directory-controller-6-2.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Pattern_Directory_Controller_6_2 class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Controller which provides REST endpoint for block patterns from wordpress.org/patterns.
+ */
+class Gutenberg_REST_Pattern_Directory_Controller_6_2 extends Gutenberg_REST_Pattern_Directory_Controller_6_0 {
+	/**
+	 * Search and retrieve block patterns metadata
+	 *
+	 * @since 5.8.0
+	 * @since 6.0.0 Added 'slug' to request.
+	 * @since 6.2.0 Added 'per_page', 'page', 'offset', 'order', and 'orderby' to request.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		/**
+		 * Include an unmodified `$wp_version`, so the API can craft a response that's tailored to
+		 * it. Some plugins modify the version in a misguided attempt to improve security by
+		 * obscuring the version, which can cause invalid requests.
+		 */
+		require ABSPATH . WPINC . '/version.php';
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		$gutenberg_data = get_plugin_data( dirname( dirname( dirname( __DIR__ ) ) ) . '/gutenberg.php', false );
+
+		$valid_query_args = array( 'offset', 'order', 'orderby', 'page', 'per_page', 'search', 'slug' );
+		$query_args       = array_merge(
+			array_intersect_key( $request->get_params(), array_flip( $valid_query_args ) ),
+			array(
+				'locale'            => get_user_locale(),
+				'wp-version'        => $wp_version, // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable -- it's defined in `version.php` above.
+				'gutenberg-version' => $gutenberg_data['Version'],
+			)
+		);
+
+		$query_args['pattern-categories'] = isset( $request['category'] ) ? $request['category'] : false;
```

Comment:
Nit: I think it's a bit more clean if we add to `$query_args` only if they have values.

Reply:
This is stripped by `$query_args = array_filter( $query_args );` — or are you're saying it would be clearer as plain if statements?